### PR TITLE
Fixes #19478 - migrate to webpack2

### PR DESCRIPTION
**Pull request by matanwerbner**
As webpack 1 is being deprecated, we better migrate to 2 to have full support and updates.

### CODE REVIEW

**Comment by matanwerbner** [comment]
storybook works fine as well.

File: package.json [hunk 1]
```diff
@@ -56,6 +56,7 @@
     "jstz": "~1.0.7",
     "lodash": "~4.15.0",
     "multiselect": "~0.9.12",
+    "fbjs": "0.8.12",
```
Comment (ohadlevy) [inline package.json]: is it required by webpack2 now?

Comment (matanwerbner) [reply package.json]: seems so.
http://stackoverflow.com/questions/43751744/cant-resolve-fbjs-lib-memoizestringonly?noredirect=1&lq=1

**Comment by ohadlevy** [comment]
[test foreman]

**Comment by ohadlevy** [comment]
[test katello]

File: package.json [hunk 2]
```diff
@@ -36,7 +35,7 @@
     "stats-webpack-plugin": "^0.2.1",
     "style-loader": "^0.13.1",
     "url-loader": "^0.5.7",
-    "webpack": "^1.9.11",
+    "webpack": "^2.5.1",
```
Comment (ohadlevy) [inline package.json]: when testing it i got:
```
error extract-text-webpack-plugin@2.1.0: The engine "node" is incompatible with this module. Expected version ">=4.3.0 < 5.0.0 || >= 5.10".
error Found incompatible module
```

any idea what are the new minimum requirements? I assume this is going to effect our build infra too.
/cc @tbrisker

Comment (ohadlevy) [reply package.json]: this is probably a local issue that i had node v5.0.0 by mistake.

**Comment by ohadlevy** [comment]
@matanwerbner I've noticed its slower, do you think wepack v2 is ready for "production"?

**Comment by matanwerbner** [comment]
@ohadlevy it is definatley pro

**Comment by matanwerbner** [comment]
Definatley prod ready although i seen many user experiencing slower build times, and reverting to v1 (which is officialy deprecated).
We can probably hold off on this for a while.

File: package.json [hunk 3]
```diff
@@ -36,7 +35,7 @@
     "stats-webpack-plugin": "^0.2.1",
```
Comment (ohadlevy) [inline package.json]: need to bump version

**Comment by ohadlevy** [comment]
also need to change the -  webpack.optimize.OccurenceOrderPlugin(), to webpack.optimize.OccurrenceOrderPlugin
but it might not be required at all anymore - see https://github.com/webpack/webpack/issues/1964 and https://gist.github.com/sokra/27b24881210b56bbaff7#occurrence-order

**Comment by ohadlevy** [comment]
@matanwerbner I've been using it over a week on my instance without a significant impact, I'm fine with merging this.

**Comment by matanwerbner** [comment]
@ohadlevy i have rebased, feel free to merge :)

File: package.json (hunk 2 above)
Comment (ohadlevy) [reply package.json]: @matanwerbner 2.6.1 is already the latest, can you revisit all the dependencies mentioned here?

**Comment by matanwerbner** [comment]
@ohadlevy tests are good

**Comment by ohadlevy** [comment]
@matanwerbner seems to work, I am however seeing the following:
```
WARNING in webpack: Using NoErrorsPlugin is deprecated.
Use NoEmitOnErrorsPlugin instead.
```

**Comment by ohadlevy** [comment]
setting as WoC as I'm still getting
```
WARNING in webpack: Using NoErrorsPlugin is deprecated.
Use NoEmitOnErrorsPlugin instead.
```

**Comment by ohadlevy** [comment]
should we wait? https://twitter.com/TheLarkInn/status/876853489049010176

**Comment by matanwerbner** [comment]
guess so... 😆

**Comment by matanwerbner** [comment]
@ohadlevy it's v3 now, no improvement in the file size, however bundle is structured a bit differently, in a way that is supposed to improve performance.
this is freshly rebased.

**Comment by ohadlevy** [comment]
nice! can you just address:
```
Chunk.modules is deprecated. Use Chunk.getNumberOfModules/mapModules/forEachModule/containsModule instead.

```

**Comment by matanwerbner** [comment]
we need to watch this space:
https://github.com/webpack-contrib/extract-text-webpack-plugin/pull/543

**Comment by ohadlevy** [comment]
I also see 
```
warning "extract-text-webpack-plugin@2.1.2" has incorrect peer dependency "webpack@^2.2.0".
```

should we bump extract-text version?

**Comment by matanwerbner** [comment]
@ohadlevy we are at latest, nowhere to bump.
extract-text-plugin does not yet support webpack v3...hence the PR i've attached.

**Comment by ohadlevy** [comment]
also..
```
warning "extract-text-webpack-plugin@2.1.0" has incorrect peer dependency "webpack@^2.2.0".
warning "sass-loader@4.1.1" has incorrect peer dependency "webpack@^2 || ^2.2.0-rc.0 || ^2.1.0-beta || ^1.12.6".
warning "stats-webpack-plugin@0.6.0" has incorrect peer dependency "webpack@^1.0||^2.1.0-beta||^2.2.0-rc".
warning "webpack-dev-server@1.16.5" has incorrect peer dependency "webpack@>=1.3.0 <3".
```

**Comment by matanwerbner** [comment]
@ohadlevy should we go back to v2, and wait for 3 to stabilize?

**Comment by ohadlevy** [comment]
On Thu, Jun 22, 2017 at 10:52 AM, matanwerbner <notifications@github.com>
wrote:

> @ohadlevy <https://github.com/ohadlevy> should we go back to v2, and wait
> for 3 to stabilize?
>

I'm running v3 in prod since your PR with no issues - is it simply a
warning that we can ignore or more than that?

**Comment by matanwerbner** [comment]
Hard to say, it seems like no braking changes have occured, i however would feel more comfortable to wait with it.
also, it doesn't bring any significant improvements to the table... your call :)

**Comment by ohadlevy** [comment]
thanks @matanwerbner

**Comment by ohadlevy** [comment]
packaging pr at https://github.com/theforeman/foreman-packaging/pull/1693

**Pull request merged**